### ISSUE TITLE
Switch statement causing compiler errors / invalid code gen in unit tests

### ISSUE THREAD

**Issue by haxiomic** [HaxeFoundation/haxe#9387]
Tested with development bec8e9e16df65dc32fcd422789595b2e9c4ed508 (Tue May 5 17:03:16 2020 +0300)

This is a weird one, it seems something in the unit tests breaks something in the haxe compiler, triggering the bug. Discovered when making this PR: https://github.com/HaxeFoundation/haxe/pull/9386

I can reproduce the issue by adding a redundant switch statement and trying to compile unit tests with `haxe compile-js.hxml`. In this case the compiler errors with `Cannot use Void as value` but in the PR example it compiled but generated code was wrong causing tests to fail

Here's the change that demonstraits the issue: a switch expression that does nothing is added to `Printer.printComplexType()`

```haxe
(wrapArgumentsInParentheses ? '($argStr)' : argStr) + " -> " + (
	/**
		@! this is the repro of the compiler issue:

		- The following expression should be equivalent to just `printComplexType(ret)` because the switch is redundant
		- When testing in simple examples everything works as expected
		- When compiling the unit tests (e.g. `haxe compile-js.hxml`), haxe fails with:
			`Cannot use Void as value`
		- If it's tweaked to compile it generates the wrong code on all targets
		- It doesn't seem to matter what we're switching on, so I've just used `true` for simplicity (see #9385 for original example)
	**/
	switch true {
		default: printComplexType(ret);
	}
);
```

I've created a branch that produces the issue using the code above
https://github.com/haxiomic/haxe/tree/cannot-use-void-repro

The commit that adds the repro: https://github.com/haxiomic/haxe/commit/602f74dfb6d5d1fc8e8ed3a762dc43d70ba0ef57

To reproduce, checkout that branch, rebuild haxe and run `haxe compile-js.hxml`, it should fail with `Cannot use Void as value`

**Comment by haxiomic** [comment]
Works on 4.0_bugfix, so _could_ be a regression or could be a recent change to the unit tests

**Comment by haxiomic** [comment]
Did a bisect on this one, making the above change to Printer.hx and re-running compile-js.hxml

Narrowed down to a change between (ok) e6de685 and (compiler-error) 27c349a
**DIff: [compare/e6de685..27c349a](https://github.com/haxeFoundation/haxe/compare/e6de685..27c349a)**

Edit: just saw recent compiler commits – guess you found this long before I did!

**Comment by haxiomic** [comment]
Re-ran bisect, this one is now fixed :D, thanks @RealyUniqueName
Fixed in this diff [compare/1c31a59..dba7789](https://github.com/haxeFoundation/haxe/compare/1c31a59..dba7789)

**Comment by RealyUniqueName** [comment]
Mystery...
